test/storage/online_file_source.cpp

```diff
@@ -409,10 +409,9 @@ TEST(OnlineFileSource, ChangeAPIBaseURL){
     util::RunLoop loop;
     OnlineFileSource fs;
     
-    loop.run();
```

Comment:
Would be great if you could split the PR in two or three commits:

- One adding the API to core
- One adding the API to android
- One adding the tests

This makes reviewing a lot easier, since reviewing is usually done "per commit". You can do this next time, but for now I'm fine squashing everything into a single commit. The last two are fixing something on the first one, so they don't really add value to the PR.